tests/unit/test_pool.py

```diff
@@ -656,7 +656,7 @@ def test_bind(self):
         for session in SESSIONS:
             session.create.assert_not_called()
             txn = session._transaction
-            txn.begin.assert_called_once_with()
```

Comment:
Since for read only txn begin will be called explicitly, we should add atleast one test case where begin txn is called explicitly

Reply:
@asthamohta This test case and file is to test the implementation for different pools. `TransactionPingingPool` has been modified to remove the begin rpc call and hence this change in tests. 

For RO transactions, test cases in `test_snapshot.py `should suffice.